File: js/src/maplestory/api/dto/characterAndroidEquipmentFaceDto.ts
import { CharacterAndroidEquipmentFaceDtoBody } from '../response/characterAndroidEquipmentDtoBody';

/**
 * 캐릭터 안드로이드 성형 정보
 */
class CharacterAndroidEquipmentFaceDto {
  /**
   * 안드로이드 성형 명
   */
  faceName: string;

  /**
   * 안드로이드 성형 베이스 컬러
   */
  baseColor: string;

  /**
   * 안드로이드 성형 믹스 컬러
   */
  mixColor: string | null;

  /**
   * 안드로이드 성형 믹스 컬러의 염색 비율
   */
  mixRate: string;

  constructor(obj: CharacterAndroidEquipmentFaceDtoBody) {
    const { face_name, base_color, mix_color, mix_rate } = obj;

    this.faceName = face_name;
    this.baseColor = base_color;
    this.mixColor = mix_color;
    this.mixRate = mix_rate;
  }
}

export { CharacterAndroidEquipmentFaceDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/character/CharacterAndroidEquipmentFaceDTO.java
package dev.spiralmoon.maplestory.api.dto.character;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

/**
 * 캐릭터 안드로이드 성형 정보
 */
@AllArgsConstructor
@Data
@ToString
public class CharacterAndroidEquipmentFaceDTO {

    /**
     * 안드로이드 성형 명
     */
    @SerializedName("face_name")
    private String faceName;

    /**
     * 안드로이드 성형 베이스 컬러
     */
    @SerializedName("base_color")
    private String baseColor;

    /**
     * 안드로이드 성형 믹스 컬러
     */
    @SerializedName("mix_color")
    private String mixColor;

    /**
     * 안드로이드 성형 믹스 컬러의 염색 비율
     */
    @SerializedName("mix_rate")
    private String mixRate;
}
